Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm

```diff
@@ -93,17 +93,23 @@
     return corePage->contextRangeForSessionWithID(sessionID);
 }
 
-std::optional<WebCore::SimpleRange> TextAnimationController::contextRangeForTextAnimationType(const WTF::UUID& uuid) const
+std::optional<WebCore::SimpleRange> TextAnimationController::contextRangeForTextAnimationID(const WTF::UUID& animationUUID) const
 {
-    if (auto sessionRange = contextRangeForSessionWithID(uuid))
+    // FIXME: This shouldn't be needed any longer, but still is.
+    if (auto sessionRange = contextRangeForSessionWithID(animationUUID))
         return sessionRange;
 
-    if (m_manuallyEnabledAnimationRanges.contains(uuid))
-        return WebCore::makeSimpleRange(m_manuallyEnabledAnimationRanges.find(uuid)->value.get());
+    if (m_manuallyEnabledAnimationRanges.contains(animationUUID))
+        return WebCore::makeSimpleRange(m_manuallyEnabledAnimationRanges.find(animationUUID)->value.get());
+
+    for (auto sessionUUID : m_initialAnimations.keys()) {
+        if (m_initialAnimations.get(sessionUUID) == animationUUID)
+            return contextRangeForSessionWithID(sessionUUID);
```

Comment:
should this map instead be keyed by animation id and have a session id value, since in this case you're trying to find a key based on a value?

Reply:
I want them all keyed the same way for now, once things settle it might make sense to optimize them in different ways.